internal/monitors/elasticsearch/elasticsearch.go

```diff
@@ -0,0 +1,119 @@
+package elasticsearch
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/signalfx/signalfx-agent/internal/core/config"
+	"github.com/signalfx/signalfx-agent/internal/monitors"
+	"github.com/signalfx/signalfx-agent/internal/monitors/elasticsearch/client"
+	"github.com/signalfx/signalfx-agent/internal/monitors/types"
+	"github.com/signalfx/signalfx-agent/internal/utils"
+	log "github.com/sirupsen/logrus"
+	"time"
+)
+
+const monitorType = "elasticsearch"
+
+var logger = log.WithFields(log.Fields{"monitorType": monitorType})
+
+// Config for this monitor
+type Config struct {
+	config.MonitorConfig `yaml:",inline" acceptsEndpoints:"true"`
+	Host                 string `yaml:"host" validate:"required"`
+	Port                 string `yaml:"port" validate:"required"`
+	// Username used to access elasticsearch stats api
+	Username string `yaml:"username"`
+	// Password used to access elasticsearch stats api
+	Password string `yaml:"password" neverLog:"true"`
+	// Protocol used to connect: http or https
+	Protocol string `yaml:"protocol" default:"http"`
```

Comment:
Just have a bool flag for `UseHTTPS` since there are only two possibilities.

Reply:
Updated